internal/auth/internal/mappers/trafficpermissionsmapper/traffic_permissions_mapper.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package trafficpermissionsmapper
+
+import (
+	"context"
+	"sync"
+
+	"github.com/hashicorp/consul/internal/auth/internal/types"
+	"github.com/hashicorp/consul/internal/controller"
+	"github.com/hashicorp/consul/internal/resource"
+	"github.com/hashicorp/consul/internal/resource/mappers/bimapper"
+	pbauth "github.com/hashicorp/consul/proto-public/pbauth/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+type TrafficPermissionsMapper struct {
+	lock     sync.Mutex
+	mapper   *bimapper.Mapper
+	liveCTPs map[string]*pbresource.ID
+}
+
+func New() *TrafficPermissionsMapper {
+	return &TrafficPermissionsMapper{
+		lock:     sync.Mutex{},
+		mapper:   bimapper.New(types.ComputedTrafficPermissionsType, types.TrafficPermissionsType),
+		liveCTPs: make(map[string]*pbresource.ID, 0),
+	}
+}
+
+//  TrafficPermissionsMapper functions
+
+func (tm *TrafficPermissionsMapper) getCTPForWorkloadIdentity(destination string) *pbresource.ID {
+	tm.lock.Lock()
+	defer tm.lock.Unlock()
+	return tm.liveCTPs[destination]
+}
+
+func (tm *TrafficPermissionsMapper) trackTPForWorkloadIdentity(tp *pbresource.ID, wi *pbresource.ID) {
+	tm.lock.Lock()
+	defer tm.lock.Unlock()
+	newTPsForWI := []resource.ReferenceOrID{tp}
+	for _, mtp := range tm.mapper.LinkIDsForItem(wi) {
+		newTPsForWI = append(newTPsForWI, mtp)
+	}
```

Comment:
One commit back, this is what I proposed, but after discussing with Eric, we decided it would be best to maintain the Item->Links immutability encouraged by the mapper.

Reply:
Ok, maybe we could add a comment explaining that in case other people have this question too.